src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageSolutionDetailControlModel.cs

```diff
@@ -162,7 +162,7 @@ protected override void CreateVersions()
             // null, if all versions are allowed to install or update
             var blockedVersions = allVersions
                 .Select(v => v.version)
-                .Where(v => !allVersionsAllowed.Any(allowed => allowed.version.Version.Equals(v)))
+                .Where(v => !allVersionsAllowed.Any(allowed => allowed.version.Equals(v)))
```

Comment:
Can you explain the change here? 
What was the type being compared? 

Ideally now the type is NuGetVersion but it's hard to tell without an editor.

Reply:
In the above code snippet `allowed.version.Version` was referring to `System.Version` reference which has it's own comparer defined and causing duplicates in allowed and blocked versions.
Referred to this [doc](https://docs.microsoft.com/en-us/dotnet/api/system.version?view=netframework-4.8#comparing-version-objects) and noticed that `System.Version` comparer is different from `NuGetVersion` comparer. 

Replaced above statement with `allowed.version` to invoke `NuGetVersion` comparer.

This issue cannot be reproduced in project PMUI because [code](https://github.com/NuGet/NuGet.Client/blob/66b48c2ee22a61dbf94bd20d7e7c392e55ecb3fc/src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageDetailControlModel.cs#L123) is already referring to `NuGetVersion` comparer while populating blocked versions list.